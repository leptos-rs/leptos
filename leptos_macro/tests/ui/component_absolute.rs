#[::leptos::component]
fn missing_scope() {}

#[::leptos::component]
fn missing_return_type(cx: ::leptos::Scope) {
    _ = cx;
}

#[::leptos::component]
fn unknown_prop_option(cx: ::leptos::Scope, #[prop(hello)] test: bool) -> impl ::leptos::IntoView {
    _ = cx; 
    _ = test;
}

#[::leptos::component]
fn optional_and_optional_no_strip(
    cx: Scope,
    #[prop(optional, optional_no_strip)] conflicting: bool,
) -> impl IntoView {
    _ = cx;
    _ = conflicting;
}

#[::leptos::component]
fn optional_and_strip_option(
    cx: ::leptos::Scope,
    #[prop(optional, strip_option)] conflicting: bool,
) -> impl ::leptos::IntoView {
    _ = cx;
    _ = conflicting;
}

#[::leptos::component]
fn optional_no_strip_and_strip_option(
    cx: ::leptos::Scope,
    #[prop(optional_no_strip, strip_option)] conflicting: bool,
) -> impl ::leptos::IntoView {
    _ = cx;
    _ = conflicting;
}

#[::leptos::component]
fn default_without_value(
    cx: ::leptos::Scope,
    #[prop(default)] default: bool,
) -> impl ::leptos::IntoView {
    _ = cx;
    _ = default;
}

#[::leptos::component]
fn default_with_invalid_value(
    cx: ::leptos::Scope,
    #[prop(default= |)] default: bool,
) -> impl ::leptos::IntoView {
    _ = cx;
    _ = default;
}

#[::leptos::component]
pub fn using_the_view_macro(cx: ::leptos::Scope) -> impl ::leptos::IntoView {
    _ = cx;
    ::leptos::view! { cx,
        "ok"
    }
}

fn main() {}
