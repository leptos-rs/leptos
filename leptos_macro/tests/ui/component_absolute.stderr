error: this method requires a `Scope` parameter
 --> tests/ui/component_absolute.rs:2:1
  |
2 | fn missing_scope() {}
  | ^^^^^^^^^^^^^^^^^^
  |
  = help: try `fn missing_scope(cx: Scope, /* ... */)`

error: return type is incorrect
 --> tests/ui/component_absolute.rs:5:1
  |
5 | fn missing_return_type(cx: ::leptos::Scope) {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: return signature must be `-> impl IntoView`

error: supported fields are `optional`, `optional_no_strip`, `strip_option`, `default` and `into`
  --> tests/ui/component_absolute.rs:10:52
   |
10 | fn unknown_prop_option(cx: ::leptos::Scope, #[prop(hello)] test: bool) -> impl ::leptos::IntoView {
   |                                                    ^^^^^

error: `optional` conflicts with mutually exclusive `optional_no_strip`
  --> tests/ui/component_absolute.rs:18:12
   |
18 |     #[prop(optional, optional_no_strip)] conflicting: bool,
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `optional` conflicts with mutually exclusive `strip_option`
  --> tests/ui/component_absolute.rs:27:12
   |
27 |     #[prop(optional, strip_option)] conflicting: bool,
   |            ^^^^^^^^^^^^^^^^^^^^^^

error: `optional_no_strip` conflicts with mutually exclusive `strip_option`
  --> tests/ui/component_absolute.rs:36:12
   |
36 |     #[prop(optional_no_strip, strip_option)] conflicting: bool,
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: unexpected end of input, expected assignment `=`
  --> tests/ui/component_absolute.rs:45:19
   |
45 |     #[prop(default)] default: bool,
   |                   ^

error: unexpected end of input, expected one of: identifier, `::`, `<`, `_`, literal, `const`, `ref`, `mut`, `&`, parentheses, square brackets, `..`, `const`

       = help: try `#[prop(default=5 * 10)]`
  --> tests/ui/component_absolute.rs:54:22
   |
54 |     #[prop(default= |)] default: bool,
   |                      ^
